mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1550,6 +1550,7 @@ def evaluate(
         self.custom_metrics = custom_metrics
         self.custom_artifacts = custom_artifacts
         self.y = dataset.labels_data
+        self.variable_mapping = self.evaluator_config.get("variable_mapping")
```

Comment:
@sunishsheth2009 Can we document this config?

Reply:
I wonder if `col_mapping` would be a better name

Reply:
Yes @prithvikannan is working on adding this in documentation: https://databricks.atlassian.net/browse/ML-35252?focusedCommentId=3966317

Reply:
> I wonder if col_mapping would be a better name

Called it variable_mapping  since in make_genai_metric, we call it variables. Do we want to change that as well? Its easier for users creating metrics using make_genai_metric and have to figure out variable_mapping instead of what col_mapping means. 🤔

Or maybe we can call it eval_fn_mapping. Col_mapping is a bit vague IMO. thoughts?